src/rebar_utils.erl

```diff
@@ -89,13 +95,128 @@ is_arch(ArchRegex) ->
         nomatch ->
             false
     end.
-
+%%
+%% REBAR_TARGET_ARCH, if used, should be set to the "standard" 
+%% target string. That is a prefix for binutils tools.
+%% "x86_64-linux-gnu" or "arm-linux-gnueabi" are good candiates
+%% ${REBAR_TARGET_ARCH}-gcc, ${REBAR_TARGET_ARCH}-ld ...
+%%
 get_arch() ->
-    Words = wordsize(),
-    otp_release() ++ "-"
-        ++ erlang:system_info(system_architecture) ++ "-" ++ Words.
+    Arch = os:getenv("REBAR_TARGET_ARCH"),
+    Words = wordsize(Arch),
+    otp_release() ++ "-" ++ get_system_arch(Arch) ++ "-" ++ Words.
+
+get_system_arch(Arch) when Arch =:= false; Arch =:= "" ->
+    erlang:system_info(system_architecture);
+get_system_arch(Arch) ->
+    Arch.
 
 wordsize() ->
+    wordsize(os:getenv("REBAR_TARGET_ARCH")).
+
+wordsize(Arch) when Arch =:= false; Arch =:= "" ->
+    native_wordsize();
+wordsize(Arch) ->
+    case match_wordsize(Arch, [{"i686","32"}, {"i386","32"},
+                               {"arm","32"}, {"aarch64", "64"},
+                               {"x86_64","64"}]) of
```

Comment:
As the list is likely to change and already spread over multiple lines, it would be easier to read if it was defined before the case...of.